File: web/src/services/syllabus.ts
export interface KnowledgePoint {
  id: string
  name: string
  requirement_raw: string
  requirement_level: string
  requirement_actions: string[]
  common_exam_style: string
  note: string
  order: number
}

export interface SyllabusSection {
  id: string
  name: string
  order: number
  knowledge_points: KnowledgePoint[]
}

export interface SyllabusChapter {
  id: string
  name: string
  order: number
  sections: SyllabusSection[]
  knowledge_points: KnowledgePoint[]
}

export interface SyllabusModule {
  id: string
  name: string
  order: number
  chapters: SyllabusChapter[]
}

export interface ExamSection {
  id: string
  name: string
  score: number | null
  duration_minutes: number | null
  description: string
  order: number
}

export interface ExamBlueprint {
  id: string
  name: string
  total_score: number | null
  duration_minutes: number | null
  description: string
  sections: ExamSection[]
}

export interface SubjectSyllabus {
  id: string
  code: string
  name: string
  order: number
  modules: SyllabusModule[]
  exam_blueprints: ExamBlueprint[]
  source_row_count: number
  knowledge_point_count: number
}

export interface SyllabusVersion {
  id: string
  source_name: string
  source_checksum: string
  row_count: number
  imported_at: string
}

export interface SyllabusTree {
  source_row_count: number
  knowledge_point_count: number
  exam_blueprint_count: number
  versions: SyllabusVersion[]
  subjects: SubjectSyllabus[]
}

export async function fetchSyllabusTree(): Promise<SyllabusTree> {
  const baseUrl = import.meta.env.VITE_API_BASE_URL ?? ''
  const response = await fetch(`${baseUrl}/api/syllabus`)

  if (!response.ok) {
    throw new Error(`Syllabus request failed with status ${response.status}`)
  }

  return (await response.json()) as SyllabusTree
}

File: backend/src/graduate_entrance/schemas/syllabus.py
from datetime import datetime

from pydantic import BaseModel


class SyllabusVersionRead(BaseModel):
    id: str
    source_name: str
    source_checksum: str
    row_count: int
    imported_at: datetime


class KnowledgePointRead(BaseModel):
    id: str
    name: str
    requirement_raw: str
    requirement_level: str
    requirement_actions: list[str]
    common_exam_style: str
    note: str
    order: int


class SectionRead(BaseModel):
    id: str
    name: str
    order: int
    knowledge_points: list[KnowledgePointRead]


class ChapterRead(BaseModel):
    id: str
    name: str
    order: int
    sections: list[SectionRead]
    knowledge_points: list[KnowledgePointRead]


class ModuleRead(BaseModel):
    id: str
    name: str
    order: int
    chapters: list[ChapterRead]


class ExamSectionRead(BaseModel):
    id: str
    name: str
    score: int | None
    duration_minutes: int | None
    description: str
    order: int


class ExamBlueprintRead(BaseModel):
    id: str
    name: str
    total_score: int | None
    duration_minutes: int | None
    description: str
    sections: list[ExamSectionRead]


class SubjectRead(BaseModel):
    id: str
    code: str
    name: str
    order: int
    modules: list[ModuleRead]
    exam_blueprints: list[ExamBlueprintRead]
    source_row_count: int
    knowledge_point_count: int


class SyllabusTreeResponse(BaseModel):
    source_row_count: int
    knowledge_point_count: int
    exam_blueprint_count: int
    versions: list[SyllabusVersionRead]
    subjects: list[SubjectRead]
